plugins/CoreHome/DataTableRowAction/RowEvolution.php

```diff
@@ -84,8 +83,14 @@ class RowEvolution
      */
     public function __construct($idSite, $date, $graphType = 'graphEvolution')
     {
+        if (!($date instanceof Date)) {
+            throw new Exception("Expected date to be an instance of \\Piwik\\Date");
+        }
```

Comment:
Would it be better or worse to use a typehint, ie, `__construct($idSite, Date $date, $graphType = '...')`?

Reply:
Thought about that, but wouldn't it throw a PHP error if a non `Date` parameter is given then?

Reply:
Yeah, that's unfortunate we can't use the language feature here... oh well.